File: BoSeonPark/투 포인터/1806.py
import sys
input = sys.stdin.readline

n, s = map(int, input().split())
elements = list(map(int, input().split()))

l, r = 0, 0
answer = 100001

tot = elements[l]
t = 1

while True:
    if tot >= s and t < answer:
        answer = t
    
    if tot < s:
        r += 1
        if r == n:
            break
        t += 1
        tot += elements[r]
    elif tot >= s:
        tot -= elements[l]
        t -= 1
        l += 1

if answer == 100001:
    print(0)
else:
    print(answer)

File: 김하영/two pointer/1806.js
const fs = require("fs");

const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";

const input = fs.readFileSync(filePath).toString().trim().split("\n");

const [N, S] = input[0].split(" ").map(Number);
const arr = input[1].split(" ").map((v) => +v);

const solution = (N, S, arr) => {
  let start = 0;
  let end = 0;
  let answer = Infinity;
  let sum = 0;

  for (start; start < N; start++) {
    // 합구하기
    while (sum < S && end < N) {
      sum += arr[end++];
    }
    if (sum >= S) {
      answer = Math.min(answer, end - start);
    }
    sum -= arr[start];
  }

  return answer === Infinity ? 0 : answer;
};

console.log(solution(N, S, arr));
